build.gradle

```diff
@@ -1,11 +1,33 @@
 plugins {
     id 'org.jetbrains.kotlin.jvm' version '1.4.0' apply false
     id 'symbol-processing' version "$ksp_version" apply false
+    id "io.gitlab.arturbosch.detekt" version "1.14.2"
 }
 
-subprojects {
+repositories {
+    jcenter()
+}
+
+allprojects {
     repositories {
         google()
         mavenCentral()
     }
+}
+
+task detektAll(type: io.gitlab.arturbosch.detekt.Detekt) {
+    parallel = true
+    failFast = false
+    buildUponDefaultConfig = true
+    source = files(projectDir)
+    config.from = project.file(".static/detekt-config.yml")
+    include("**/*.kt")
```

Comment:
are these include/excludes needed? I don't remember needing them.

Reply:
Yep, you're right.

Reply:
heh, yeah

Reply:
Actually, sorry, I think the `include` is still needed, otherwise, stuff like this is outputted:
```
...
Ignoring a file detekt cannot handle: .../kotlin-inject/integration-tests/module/build.gradle
Ignoring a file detekt cannot handle: .../kotlin-inject/integration-tests/ksp-companion/build.gradle
Ignoring a file detekt cannot handle: .../kotlin-inject/integration-tests/kapt/build.gradle
Ignoring a file detekt cannot handle: .../kotlin-inject/integration-tests/kapt-companion/build.gradle
Ignoring a file detekt cannot handle: .../kotlin-inject/LICENSE
...
```

Reply:
Looking at my other project, you can set the source to the sourceSet explicitly, that should be cleaner than wildcards imo https://github.com/willowtreeapps/assertk/blob/cf0d1da83c71130a12f81b0665d4faf68fc2eb1b/build.gradle#L72

Reply:
Thanks for the reference. It does look interesting but I have to note that that approach has a slight inconvenience of needing to update the detekt task config when a new module/source set is added if I understand correctly. Or more like needing to _not forget_ to do that.